superset/models/core.py

```diff
@@ -809,12 +810,29 @@ def needs_conversion(df_series):
                 return True
             return False
 
+        username = utils.get_username()
```

Comment:
(optional): was trying to think of a way to duplicate less lines of code, maybe an inline function definition above the with block:
```python
def _log_query(sql):
    if log_query:
        log_query(
            engine.url,
            sql,
            schema,
            username,
            'superset.models.core',
)
```
then the 2 calls in the with block would be one-liners